main.cpp

```diff
@@ -0,0 +1,222 @@
+#include <iostream>
+#include <fstream>
+#include <string.h>
+#include "./header.h"
+
+using std::cout;
+using std::endl;
+using std::string;
+using std::ifstream;
+using std::ofstream;
+
+const string OUTPUT_FILE_PATH = "output.txt";
+
+int main(int argc, char *argv[]) {
+
+    string data_file_path = "";
+    // Make sure that the argument isn't empty
+    if (argv[1] != nullptr) {
+
+        data_file_path = static_cast<string>(argv[1]);
+
+        if (argv[2] != nullptr) {
+            string argument2 = static_cast<string>(argv[2]);
+
+            if (argument2 == "search") {
+
+                if (argv[3] != nullptr) {
+                    string search_string = static_cast<string>(argv[3]);
+
+                    string_search(data_file_path, search_string);
+                } else {
+                    cout << "Argument 3 invalid.";
+
+                }
+
+            } else if (argument2 == "sort") {
+
+                if (argv[3] != nullptr) {
+                    string sort_order = static_cast<string>(argv[3]);
+
+                    string_sort(data_file_path, sort_order);
+
+
+                } else {
+                    cout << "Argument 3 invalid. You must use asc or desc." << endl;
+                }
+
+            } else if (argument2 == "count") {
+
+                string_count(data_file_path);
+
+            } else {
+                cout << "Argument 2 invalid." << endl;
+            }
+
+
+        }
+    }
+
+    else cout << "No arguments provided.";
+    return 0;
+}
+
+int string_search(string data_file_path, string search_string){
+
+    int line_number = 0;
+    int line_position = 0;
+    ifstream data_file;
+    ofstream output_file;
+    data_file.open(data_file_path);
+    output_file.open("./Search_" + OUTPUT_FILE_PATH);
+
+    output_file << "SEARCH" << endl;
+            output_file << "Searched for: " << search_string << endl;
+    output_file << "Located on:" << endl;
+    output_file << "Pos\t\tLine" << endl;
+    if (data_file.is_open()) {
+        string read_line;
+        while (std::getline(data_file, read_line)) {
+
+            line_position = read_line.find(search_string, 0);
+            line_number++;
+
```

Comment:
I was trying to do this same thing for search, but was having trouble.  I ended up following your lead on this one.  This was easy to understand code!